contracts/chainlink/ChainLink.sol

```diff
@@ -21,7 +21,7 @@ contract ChainLink is ChainlinkClient, Oracle {
 
     function pack(uint128 pmt_, ERC20 tok_) { // payment
         require(uint32(now) > lag);
-        link.transferFrom(msg.sender, address(this), uint(pmt_));
+        require(link.transferFrom(msg.sender, address(this), uint(pmt_)));
```

Comment:
Here a failed `transferFrom` could result in setting `pmt` and `lag` allowing an attacker to take advantage of additional reserves that haven't been used by the oracle, and getting rewarded for it.